src/menu/components/boxart.c

```diff
@@ -65,15 +65,36 @@ void component_boxart_free (component_boxart_t *b) {
 }
 
 void component_boxart_draw (component_boxart_t *b) {
-    if (b && b->image && b->image->width == BOXART_WIDTH && b->image->height == BOXART_HEIGHT) {
+    if (b && b->image  && b->image->width <= BOXART_WIDTH_MAX && b->image->height <= BOXART_HEIGHT_MAX) {
         rdpq_mode_push();
             rdpq_set_mode_copy(false);
-            rdpq_tex_blit(
-                b->image,
-                BOXART_X,
-                BOXART_Y,
-                NULL
-            );
+            if(b->image->height == BOXART_HEIGHT_MAX)
+            {
+                rdpq_tex_blit(
+                    b->image,
+                    BOXART_X_JP,
+                    BOXART_Y_JP,
+                    NULL
+                );
+            }
+            else if(b->image->width == BOXART_WIDTH_DD && b->image->height == BOXART_HEIGHT_DD )
+            {
+                rdpq_tex_blit(
+                    b->image,
+                    BOXART_X_DD,
+                    BOXART_Y_DD,
+                    NULL
+                );
+            }
+            else
+            {
+                rdpq_tex_blit(
+                    b->image,
+                    BOXART_X,
+                    BOXART_Y,
+                    NULL
+                );
+            }
```

Comment:
this can be done much more elegant - use `b->image->width` and `b->image->height` to calculate the position of the top left corner, and use single constant `BOXART_X/Y` to anchor the bottom right corner

Reply:
this way picture with arbitrary size and aspect ratio would be correctly placed in the bottom right corner of the screen